aldryn_search/search_indexes.py

```diff
@@ -43,8 +43,49 @@ def get_plugin_queryset(self, language):
         return queryset
 
     def get_search_data(self, obj, language, request):
+        """
+        In the project settings set up the variable
+
+        PLACEHOLDERS_SEARCH_LIST = {
+            '*': ['slot1', 'slot2', etc. ], #mandatory
+            'reverse_id_alpha': ['slot1', 'slot2', etc. ],
+            'reverse_id_beta': ['slot1', 'slot2', etc. ],
+            # exclude it from the placehoders search list
+            # (however better to remove at all to exclude it)
+            'reverse_id_empty': []
+            etc.
+        }
+
+        Put '-' (minus) before the slot string to exclude it
+
+        'slot_name' -> search for slot_name
+        '-slot_name' -> exclude slot name from the search list
+
+        or leave it empty
+
+        PLACEHOLDERS_SEARCH_LIST = {}
+        """
         current_page = obj.page
-        placeholders = current_page.placeholders.all()
+        reverse_id = current_page.reverse_id
```

Comment:
please move the logic from here all the way to the placeholder queryset to a method called get_page_placeholders, which takes a page as parameter and returns placeholder queryset after doing the filtering logic